### PR TITLE
CEDS-1858 Switch Client ID depending on User Agent

### PR DESCRIPTION
**Pull request by ed0906**
In MDTP environments requests always send with a `User-Agent` header
We can use this to switch between the Movements & Internal UI Client IDs (they both need their own)

Annoyingly when running locally this header isn't sent so there is a fallback which can be blanked out in non-local environments

### CODE REVIEW

**Comment by ed0906** [comment]
test this please

File: test/unit/uk/gov/hmrc/exports/movements/connector/CustomsInventoryLinkingExportsConnectorSpec.scala [hunk 1]
```diff
@@ -16,61 +16,52 @@
...
-import uk.gov.hmrc.http.logging.Authorization
-import unit.uk.gov.hmrc.exports.movements.base.{CustomsExportsBaseSpec, MockHttpClient}
 
-import scala.concurrent.Future
-import scala.util.Random
-import scala.xml.{Elem, NodeSeq}
+import scala.xml.Elem
 
-class CustomsInventoryLinkingExportsConnectorSpec extends CustomsExportsBaseSpec {
+class CustomsInventoryLinkingExportsConnectorSpec extends ConnectorSpec {
```
Comment (PatrykRudnicki) [inline test/unit/uk/gov/hmrc/exports/movements/connector/CustomsInventoryLinkingExportsConnectorSpec.scala]: If this test is integration test and ConnectorSpec contains application please move it outside package `unit`

File: app/uk/gov/hmrc/exports/movements/config/AppConfig.scala [hunk 2]
```diff
@@ -32,10 +35,16 @@ class AppConfig @Inject()(runModeConfiguration: Configuration, servicesConfig: S
     throw new IllegalStateException("Missing configuration for Customs Inventory Linking send arrival URI")
   )
 
-  lazy val clientIdInventory = servicesConfig.getConfString(
-    "customs-inventory-linking-exports.client-id",
-    throw new IllegalStateException("Missing configuration for Customs Inventory Linking Client Id")
-  )
+  def clientIdInventory(implicit hc: HeaderCarrier): String = {
```
Comment (bambuchaAdm) [inline app/uk/gov/hmrc/exports/movements/config/AppConfig.scala]: Can we move this as single loaded `Map[String, String]` with default value ?

File: conf/application.conf [hunk 3]
```diff
@@ -160,7 +160,9 @@ microservice {
             host = localhost
             port = 9823
             sendArrival = "/"
-            client-id = "5c68d3b5-d8a7-4212-8688-6b67f18bbce7"
+            client-id {
+                default = "5c68d3b5-d8a7-4212-8688-6b67f18bbce7"
```
Comment (bambuchaAdm) [inline conf/application.conf]: What if received client-id would be default ?

Comment (ed0906) [reply conf/application.conf]: Not sure I understand

File: app/uk/gov/hmrc/exports/movements/config/AppConfig.scala (hunk 2 above)
Comment (ed0906) [reply app/uk/gov/hmrc/exports/movements/config/AppConfig.scala]: I see no helper methods for getting Maps out of config, do you know of one?

**Pull request merged**